### PR TITLE
Added Nigthvision

### PR DESCRIPTION
**Pull request by jorgun**
## About the PR

copied nigthvision from https://github.com/Rxup/space-station-14
changed some of it. 

## Technical details

added nigthvision to:
ninja visors
engingering goggels
Syndicate and Deathsquad gasmasks
some animals that shud have good nigthvision.
dragon and carps

## Media

nightvision disabled
image
nightvision enabled
image
nightvision enabled on animal
image

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- add: Added Nigthvision

## Release Notes

- **New Features**
	- Introduced night vision capabilities with new components and systems for enhanced gameplay.
	- Added toggleable night vision effects for various entities, including players and NPCs.
	- New night vision items and actions, enhancing visibility in low-light conditions.

- **Bug Fixes**
	- Improved functionality and responsiveness of night vision systems and overlays.

- **Documentation**
	- Updated item and entity prototypes to include new night vision attributes and functionalities.

### CODE REVIEW

**Comment by BombasterDS** [comment]
Finally someone port nightvision, thanks!
But i think you should remove nightvision from command hardsuits because command players will always walk in hardsuits just to see better in the dark. Also add nightvision to nukies visor instead of hardsuits.

**Review by whateverusername0 — CHANGES_REQUESTED** [review]
looks good but

rename PNV to NVG
also remove NVG from heads/cap hardsuits
move it to the syndicate gasmask, ninja visor and mesons (engi goggles)

**Comment by Piras314** [comment]
I feel like it really needs some shader or at least noise texture over it.

**Comment by JoeHammad1844** [comment]
I think the night vision should be moved from the syndicate gas mask to the visor

File: Content.Client/Backmen/Eye/NightVision/NightVisionSystem.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+using Content.Client.Overlays;
+using Content.Shared.GameTicking;
+using Content.Shared.Backmen.Eye.NightVision.Components;
+using Content.Shared.Inventory.Events;
+using Robust.Client.Graphics;
+using Robust.Client.Player;
+using Robust.Shared.Player;
+
+namespace Content.Client.GG.Eye.NightVision;
```
Comment (whateverusername0) [inline Content.Client/Backmen/Eye/NightVision/NightVisionSystem.cs:9]: no GG namespace x2

File: Content.Shared/Backmen/Eye/NightVision/Components/PNVComponent.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+using Content.Shared.Actions;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Shared.Backmen.Eye.NightVision.Components;
+
+
+[RegisterComponent, NetworkedComponent]
+public sealed partial class PNVComponent : Component
```
Comment (whateverusername0) [inline Content.Shared/Backmen/Eye/NightVision/Components/PNVComponent.cs:10]: rename PNV to NVD or NVG

**Review by whateverusername0 — CHANGES_REQUESTED** [review]

**Comment by Piras314** [comment]
Visually two changes would be great:
1. Some kind of noise or otherwise some kind of shader
2. The ability to set different colours for the visor, for example the syndicate gas mask being red would be very interesting. Green and red would probably be the only colours though.

**Comment by a-person5660** [comment]
i would love to see some white phosphorus NVGs. 

also give changelings this when they use that enhanced eyesight ability thingy

**Comment by whateverusername0** [comment]
> also give changelings this when they use that enhanced eyesight ability thingy

they gain thermals too :godo:

**Comment by jorgun** [comment]
im redoing some of the code. 
and whit all the rewrite im gonna move it into the /goobstation/ structure. 

there is problem whit having it add the comp and removing it to the mob. gonn look into making it more contained to the item.
i dont know how to add static overlay, but i like the idea. will work on that when. other stuff is stable.
i gonna need feedback from admin on what items and animals to add this to.

**Comment by a-person5660** [comment]
👍

**Comment by JoeHammad1844** [comment]
> they gain thermals too :godo:

speaking of thermals, the bulldog bundle gets a pair in their bag, when they're implemented

**Comment by ElPidor** [comment]
it would be cool being red vision for syndicate gasmask since they have red visor 
For colour blind people theyll see it gray so its good 
Imho could this pr rule to future RP thermal vision update?

**Comment by Piras314** [comment]
remember it's _Goobstation now

**Comment by a-person5660** [comment]
is this still being worked on?

**Comment by JoeHammad1844** [comment]
it's over..

**Comment by jorgun** [comment]
i git a bit to abisus whit this one, i managed to make it work as a raseial ability for dragon cats, but for NVG goggels i diddnt manage to get it working.

**Pull request closed without merging**